Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -579,43 +579,92 @@ u16 GCMemcard::DEntry_BlockCount(u8 index) const
   return blocks;
 }
 
-u32 GCMemcard::DEntry_CommentsAddress(u8 index) const
+std::optional<std::vector<u8>> GCMemcard::GetSaveDataBytes(u8 save_index, size_t offset,
+                                                           size_t length) const
 {
-  if (!m_valid || index >= DIRLEN)
-    return 0xFFFF;
+  if (!m_valid || save_index >= DIRLEN)
+    return std::nullopt;
 
-  return GetActiveDirectory().m_dir_entries[index].m_comments_address;
-}
+  const DEntry& entry = GetActiveDirectory().m_dir_entries[save_index];
+  const BlockAlloc& bat = GetActiveBat();
+  const u16 block_count = entry.m_block_count;
+  const u16 first_block = entry.m_first_block;
+  const u32 block_max = BAT_SIZE + MC_FST_BLOCKS;
```

Comment:
```suggestion
  constexpr u32 block_max = BAT_SIZE + MC_FST_BLOCKS;
```

Reply:
Actually, looking at that again, I probably want `block_max` to be the maximum block of the current memory card, adjusting for actual card size, not the theoretical maximum.